### DISCUSSION TITLE
Is there a way to compare two definitions in a condition?

### DISCUSSION THREAD

**Discussion by kelsob** [dialogic-godot/dialogic#860] [Q&A] [answered]
Basically the title, dialogic is great and I'm having a lot of fun working with it but I can't figure out any built in way of making a comparison of two definitions in a condition event. So far I can only see the ability to compare a definition with a hard-coded value in the condition event, but not another definition. I can do: Does [X] = 5, but not does [X] = [Y] in essence. 

 I'm not fluent yet enough with tweaking the source code to add this functionality without horribly breaking dialogic. Any help on this front would be extremely helpful. I realize there are workarounds that involve making these comparisons outside of dialogic itself, but it seems like such a simple level of additional functionality that I feel like I must be missing something.

**Comment by Jowan-Spooner** [comment, marked as answer]
You can write the name of a definition into the input field and it will use the value of that definition. 

This works fine unless you have multiple definitions with the same name, as you cannot use the path currently. This is a known issue and will hopefully be changed, but not a priority atm. 

Hope this helps.

**Reply by kelsob** [reply]
Thanks so much, knowing it was possible I was able to figure out what I was doing wrong. 
For anyone else who may be curious if they run into a similar problem, in the condition I was attempting to reference the definitions with square brackets the same way you would inside of a Text event. Obviously just doesn't work this way, just plain text reference to the variable cleared up my issue: Variable vs. [Variable]. Thanks again for the swift fix 👍

**Reply by Jowan-Spooner** [reply]
Yeah having it in square brackets would make more sense anyways. This is a pretty old feature, and I think it's likely gonna be replaced by a more clean way of doing it in 2.X, but who knows when that will be.
